src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -502,9 +499,10 @@ SHAMap::onlyBelow (SHAMapTreeNode* node) const
 
     // An inner node must have at least one leaf
     // below it, unless it's the root_
-    assert (node->hasItem () || (node == root_.get ()));
+    auto leaf = static_cast<SHAMapTreeNode*>(node);
+    assert (leaf->hasItem () || (leaf == root_.get ()));
```

Comment:
I could be wrong that `root_` can sometimes be a leaf.  But here:
https://github.com/ripple/rippled/pull/1111/files#diff-cdf9517dc2cc23f526612bcaf5d27918L981
is existing code that asks that very question.  So I just assumed it could.